flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/source/row/converter/PostgresRowConverter.java

```diff
@@ -28,4 +34,39 @@
 	public PostgresRowConverter(RowType rowType) {
 		super(rowType);
 	}
+
+	@Override
+	public JDBCFieldConverter createConverter(LogicalType type) {
+		LogicalTypeRoot root = type.getTypeRoot();
+
+		if (root == LogicalTypeRoot.ARRAY) {
+			ArrayType arrayType = (ArrayType) type;
+			LogicalTypeRoot elemType = arrayType.getElementType().getTypeRoot();
+
+			if (elemType == LogicalTypeRoot.VARBINARY) {
+
+				return v -> {
+					PgArray pgArray = (PgArray) v;
+					Object[] in = (Object[]) pgArray.getArray();
+
+					Object[] out = new Object[in.length];
+					for (int i = 0; i < in.length; i++) {
+						out[i] = ((PGobject) in[i]).getValue().getBytes();
+					}
+
+					return out;
+				};
+			} else {
+				return v -> ((PgArray) v).getArray();
```

Comment:
Should we add the default conversion for ARRAY in `AbstractJDBCRowConverter`? Currently, we will directly put `java.sql.Array` into the Row which is not correct.

Reply:
>Should we add the default conversion for ARRAY in AbstractJDBCRowConverter? 
>Currently, we will directly put java.sql.Array into the Row which is not correct.

There's no default array conversion yet. In pg, it's always need to case to `PgArray`. 

What it puts in a Row is primitive array (obtained from PgArray.getArray()), not java.sql.Array

Reply:
I mean the currently default implementation of `AbstractJDBCRowConverter` will use `v -> v` for the array conversion, which puts `java.sql.Array` in a Row.

Reply:
gotcha. 

AbstractJDBCRowConverter is migrated from JDBCInputFormat, It's always been like that in JDBCInputFormat. We can explicitly disable array in AbstractJDBCRowConverter in a separate jira if you want.